system-functionalities.md

```diff
@@ -141,7 +144,7 @@ Protocol:
     1. Outputs a success indicator.
 1. The client:
     1. Closes the session.
-    1. Outputs `key_id` to the calling application.
+    1. Outputs `key_id` and `StorageObject` to the calling application.
```

Comment:
Pending the first question, if the `Secret` or `StoredObject` type already has the `key_id` in it, why do we need to return it separately from the `StoredObject`?

Reply:
Because the `StoredObject` likely won't remain directly readable by the calling application.

Reply:
i.e., all these adjustments are just placeholders so that the protocol and memory-handling are more likely to remain correct as we iterate on local storage.

Reply:
Ah that makes sense. So these the objects (StorageObject and KeyId) should still be returned as a tuple and not one singular object?

Reply:
Yes, that's right.